cloudinit/net/__init__.py

```diff
@@ -19,6 +20,19 @@
 LOG = logging.getLogger(__name__)
 SYS_CLASS_NET = "/sys/class/net/"
 DEFAULT_PRIMARY_INTERFACE = 'eth0'
+OVS_INTERNAL_INTERFACE_LOOKUP_CMD = [
+    "ovs-vsctl",
+    "--format",
+    "csv",
+    "--no-headings",
+    "--timeout",
+    "10",
+    "--columns",
+    "name",
+    "find",
+    "interface",
+    "type=internal",
```

Comment:
How confident are we that this query will find the intersection of system devices and ovs devices?  I read some of the bug discussion; however I'm wondering if we want to enumerate all ovs devices and skip the intersection of system devices and ovs devices?  Would that help future proof from some other ovs interface not type=internal but an interface we need to skip?

Reply:
> How confident are we that this query will find the intersection of system devices and ovs devices? I read some of the bug discussion; however I'm wondering if we want to enumerate all ovs devices and skip the intersection of system devices and ovs devices?

Part of the issue is that it isn't clear to me that we _only_ want to consider devices that OVS has as type=system. Consider `bond0` in an instance launched by the integration test from this PR:

```
# ovs-vsctl --columns=name,type find interface type!=internal
name                : bond0
type                : ""
```

While we will skip `bond0` for other reasons (it _is_ a bond per `/sys/class/net`), it's not clear to me that we want to reject it _because of its state in OVS_.

Expressed differently: I'm not sure what classes of interface will have a type that isn't "internal" or "system" but, in the configurations I've seen, "type=internal" skips every interface that isn't presented "normally" through `/sys/class/net`. I think those are the ones we need to exclude.

> Would that help future proof from some other ovs interface not type=internal but an interface we need to skip?

Potentially, but it could also mean that we skip a class of interfaces that we _shouldn't_. Given that we don't _know_, I'm erring on the side of caution for now: only exclude interfaces that we're sure should be excluded.

Reply:
I'm fine with the cautious approach.  If we find out that there are additional "internal" interfaces that aren't listed in the current query we can look at a different approach of enumeration/filtering.